src/raftstore/store/fsm/peer.rs

```diff
@@ -940,15 +959,36 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         // the size of current CompactLog command can be ignored.
         let remain_cnt = peer.last_applying_idx - state.get_index() - 1;
         peer.raft_log_size_hint = peer.raft_log_size_hint * remain_cnt / total_cnt;
-        let task = RaftlogGcTask {
+        let task = BaseRaftLogGcTask {
             raft_engine: Arc::clone(&peer.get_store().get_raft_engine()),
             region_id: peer.get_store().get_region_id(),
             start_idx: peer.last_compacted_idx,
             end_idx: state.get_index() + 1,
         };
         peer.last_compacted_idx = task.end_idx;
-        peer.mut_store().compact_to(task.end_idx);
-        if let Err(e) = self.raftlog_gc_worker.schedule(task) {
+        {
+            let last_gc_vec_len = self.entry_cache_gc_handler.last_compact_gc_vec_len;
+            let mut gc_vec = Vec::with_capacity(last_gc_vec_len * 11 / 10);
```

Comment:
Is this a magic number? Any reason for `* 1.1`?

Reply:
Just like the default parameter which used by hash table to recalculate capacity. During stress test, the numbers of entry cache needed to be reclaimed are similar. I think the difference between current the previous to may be under 10%. 😂